File: app/src/main/java/com/whoguri/learnmvp/javaBase/PreferenceManager.java
package com.whoguri.learnmvp.javaBase;

import android.app.Activity;
import android.content.SharedPreferences;

/* whoGuri 13/11/18 */
public class PreferenceManager {
    SharedPreferences preferences;
    SharedPreferences.Editor editor;

    public PreferenceManager(Activity activity) {
        preferences = activity.getSharedPreferences("pref", android.content.Context.MODE_PRIVATE);
        editor=preferences.edit();
    }

    void setStringValue(String key, String value ) {
        editor.putString(key,value);
        editor.commit();
    }

    String getStringValue(String key){
        return preferences.getString(key,"");
    }
    void clearPref(){
        editor.clear().commit();
    }
}

File: app/src/main/java/com/whoguri/learnmvp/kotlinBase/PreferenceManager.kt
package com.whoguri.learnmvp.kotlinBase

import android.content.SharedPreferences
import android.support.v7.app.AppCompatActivity

/* whoGuri 13/11/18 */
class PreferenceManager(val activity: AppCompatActivity) {
    var preferences: SharedPreferences;
    var editor: SharedPreferences.Editor;

    init {
        preferences = activity.getSharedPreferences("pref", android.content.Context.MODE_PRIVATE);
        editor = preferences.edit();
    }

    fun setStringValue(key: String, value: String) {
        editor.putString(key, value);
        editor.commit();
    }

    fun getStringValue(key: String) = preferences.getString(key, "")
    fun clearPref() {
        editor.clear().commit();
    }
}
